### PR TITLE
Helpers to compute intermediate outputs

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: docs/examples/plot_intermediate_outputs.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+We reuse the pipeline implemented in example
+`Column Transformer with Mixed Types
+<https://scikit-learn.org/stable/auto_examples/compose/plot_column_transformer_mixed_types.html#sphx-glr-auto-examples-compose-plot-column-transformer-mixed-types-py>`_.
+There is one change because
+`ONNX-ML Imputer <https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.Imputer>`_
+does not handle string type. This cannot be part of the final ONNX pipeline
+and must be removed. Look for comment starting with ``---`` below.
+"""
+import pandas as pd
```
Comment (prabhat00155) [inline docs/examples/plot_intermediate_outputs.py]: Alphabetical order.

File: docs/examples/plot_intermediate_outputs.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+preprocessor = ColumnTransformer(
+    transformers=[
+        ('num', numeric_transformer, numeric_features),
+        ('cat', categorical_transformer, categorical_features),
+        ])
+
+clf = Pipeline(steps=[('preprocessor', preprocessor),
+                      ('classifier', LogisticRegression(solver='lbfgs'))])
+
```
Comment (prabhat00155) [inline docs/examples/plot_intermediate_outputs.py]: There should be zero or one blank line here. Two blank lines should be used to surround top-level functions and class definitions.

File: docs/examples/plot_intermediate_outputs.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        elif v == 'float64':
+            t = FloatTensorType([1, 1])
+        else:
+            t = StringTensorType([1, 1])
+        inputs.append((k, t))
+    return inputs
+    
+inputs = convert_dataframe_schema(X_train)
+
+import pprint
```
Comment (prabhat00155) [inline docs/examples/plot_intermediate_outputs.py]: Imports should be at the beginning of a file. Please make changes at other places as well.

File: skl2onnx/helpers/onnx_helper.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    """
+    if isinstance(onnx_file_or_bytes, str):
+        with open(onnx_file_or_bytes, "rb") as f:
+            return onnx.load(f)
+    elif hasattr(onnx_file_or_bytes, 'read'):
+        return onnx.load(onnx_file_or_bytes)
+    else:
+        b = BytesIO(onnx_file_or_bytes)
+        return onnx.load(b)
+
```
Comment (prabhat00155) [inline skl2onnx/helpers/onnx_helper.py]: Two blank lines not three.

File: skl2onnx/helpers/onnx_helper.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    onnx_model.producer_name = model.producer_name
+    onnx_model.producer_version = model.producer_version
+    onnx_model.domain = model.domain
+    onnx_model.model_version = model.model_version
+    onnx_model.doc_string = model.doc_string
+    
+    if len(onnx_model.graph.input) != len(model.graph.input):
+        raise RuntimeError("Input mismatch {} != {}".format(len(onnx_model.input), len(model.input)))
+    return onnx_model
+
```
Comment (prabhat00155) [inline skl2onnx/helpers/onnx_helper.py]: Why this extra blank line?

File: docs/examples/plot_intermediate_outputs.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+# We are ready to run *onnxruntime*.
+
+import onnxruntime as rt
+import numpy
+sess = rt.InferenceSession("pipeline_titanic.onnx")
+pred_onx = sess.run(None, inputs)
+print("predict", pred_onx[0][:5])
+print("predict_proba", pred_onx[1][:1])
+
+
```
Comment (prabhat00155) [inline docs/examples/plot_intermediate_outputs.py]: One blank line not two.

File: docs/examples/plot_intermediate_outputs.py (hunk 3 above)
Comment (xadupre) [reply docs/examples/plot_intermediate_outputs.py]: I usually write examples differently as I write code. I import module for the first time they are needed in the example.

**Pull request merged**